File: Android/feature-dreams-impl/src/main/java/ru/iandreyshev/featureDreams/storage/IDreamsStorage.kt
package ru.iandreyshev.featureDreams.storage

import io.reactivex.Observable
import ru.iandreyshev.featureDreams.storage.entity.DreamStorageEntity
import ru.iandreyshev.featureDreamsApi.domain.DreamKey

interface IDreamsStorage {
    val dreamObservable: Observable<List<DreamStorageEntity>>
    fun save(dream: DreamStorageEntity)
    fun save(dreams: List<DreamStorageEntity>)
    fun delete(key: DreamKey)
    fun clear()
}

File: Backend/Server/Storage/Dreams/IDreamsStorage.cs
using Dreamland.Domain;
using System.Collections.Generic;

namespace Dreamland.Storage.Dreams
{
	public interface IDreamsStorage : IStorage
	{
		List<Dream> All(long userId);
		long Add(Dream dream);
		void Update(Dream dream);
		void Delete(long dreamId);
		void DeleteAllForUser(long userId);
	}
}
